staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go

```
@@ -215,15 +286,22 @@ func (o *AutoscaleOptions) Run() error {
 		mapping := info.ResourceMapping()
 		gvr := mapping.GroupVersionKind.GroupVersion().WithResource(mapping.Resource.Resource)
 		if _, err := o.scaleKindResolver.ScaleForResource(gvr); err != nil {
-			return fmt.Errorf("cannot autoscale a %v: %v", mapping.GroupVersionKind.Kind, err)
+			return fmt.Errorf("cannot autoscale a %s: %w", mapping.GroupVersionKind.Kind, err)
 		}
 
 		// handles the creation of HorizontalPodAutoscaler objects for both v2 and v1 APIs.
 		// If v2 API fails, try to create and handle HorizontalPodAutoscaler using v1 API
-		hpaV2 := o.createHorizontalPodAutoscalerV2(info.Name, mapping)
+		//
+		// Note: In the validateResourceTarget method, we've already validated the resource target configurations,
+		// Therefore, we do not return the error but proceed with fallback logic.
+		hpaV2, _ := o.createHorizontalPodAutoscalerV2(info.Name, mapping)
```

Either don't introduce that error returned from `createHorizontalPodAutoscalerV2` or make sure to act on it. Swallowing errors is bad.

---

Modified to handle this error.